src/rpc/blockchain.cpp

```diff
@@ -809,6 +808,28 @@ CoinStatsHashType ParseHashType(const std::string& hash_type_input)
     }
 }
 
+//! Calculate statistics about the unspent transaction output set
+
+std::optional<kernel::CCoinsStats> GetUTXOStats(CCoinsView* view, node::BlockManager& blockman,
+                                                kernel::CoinStatsHashType hash_type,
+                                                const std::function<void()>& interruption_point = {},
+                                                const CBlockIndex* pindex = nullptr,
+                                                bool index_requested = true)
+{
+    // Use CoinStatsIndex if it is requested and available and a hash_type of Muhash or None was requested
+    if ((hash_type == kernel::CoinStatsHashType::MUHASH || hash_type == kernel::CoinStatsHashType::NONE) && g_coin_stats_index && index_requested) {
+        if (pindex) {
+            return g_coin_stats_index->LookUpStats(pindex);
+        } else {
+            CBlockIndex* block_index = WITH_LOCK(cs_main, return blockman.LookupBlockIndex(view->GetBestBlock()));
+            return g_coin_stats_index->LookUpStats(block_index);
+        }
+    }
+
+    auto hasher = kernel::MakeUTXOHasher(hash_type);
+    return kernel::GetUTXOStatsWithHasher(*hasher, view, blockman, interruption_point);
```

Comment:
Maybe we could make an even clearer naming distinction here, I think. While we can really only "get" the stats from the index the kernel actually computes them. So the kernel function could be renamed to `ComputeUTXOStatsWithHasher`. But it's fine to leave it as is. I just found it an interesting thought and interested to hear what others think.

Reply:
Fixed as of push 8b853034a13f49b85cdf4168d9d49bd297f5b8ce